src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,69 @@
+﻿module Deciders.BankAccount
+
+open System
+open BankAccountPrimitives
+
+type BankAccountOpened =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: AccountId
+      Amount: decimal
+      CashierId: CashierId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: AccountId
+      Amount: decimal
+      ATMId: AtmId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: AccountId
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
+
+type Event =
+    | BankAccountOpened of BankAccountOpened
+    | DepositRecorded of DepositRecorded
+    | CashWithdrawnFromATM of CashWithdrawnFromATM
+    | BankAccountClosed of BankAccountClosed
+
+type BankAccount =
+    | NotInitialised
+    | Open of
+        {| Id: AccountId
```

Comment:
ids should not be part of the state, If you make it multitenant you won't put the tenantId in there either

Reply:
I'll think of that, but in the other PR 🙂